src/pipeline.rs

```diff
@@ -21,66 +26,149 @@ pub fn build_graphs(
     config: &Configuration,
     in_memory_entity_mapping_persistor: Arc<InMemoryEntityMappingPersistor>,
 ) -> Vec<SparseMatrix> {
-    let sparse_matrices = create_sparse_matrices(&config.columns);
-    dbg!(&sparse_matrices);
+    let sparse_matrices_desc = Arc::new(create_sparse_matrices_descriptors(&config.columns));
+    dbg!(&sparse_matrices_desc);
 
-    let mut bus: Bus<SmallVec<[u64; SMALL_VECTOR_SIZE]>> = Bus::new(128);
-    let mut sparse_matrix_threads = Vec::new();
-    for mut sparse_matrix in sparse_matrices {
-        let rx = bus.add_rx();
-        let handle = thread::spawn(move || {
-            for received in rx {
-                sparse_matrix.handle_pair(&received);
-            }
-            sparse_matrix.finish();
-            sparse_matrix
-        });
-        sparse_matrix_threads.push(handle);
-    }
+    let node_indexer_builders: Vec<Box<NodeIndexerBuilder>> = sparse_matrices_desc
+        .iter()
+        .map(|md| Box::new(NodeIndexerBuilder::new(md)))
+        .collect();
 
-    for input in config.input.iter() {
-        let mut entity_processor = EntityProcessor::new(
-            config,
-            in_memory_entity_mapping_persistor.clone(),
-            |hashes| {
-                bus.broadcast(hashes);
-            },
-        );
-
-        match &config.file_type {
-            FileType::Json => {
-                let mut parser = dom::Parser::default();
-                read_file(input, config.log_every_n as u64, move |line| {
-                    let row = parse_json_line(line, &mut parser, &config.columns);
-                    entity_processor.process_row(&row);
-                });
+    let processing_worker_num = num_cpus::get();
```

Comment:
I think it's worth adding this to application args and seting default to num_cpus::get(). It may be useful to limit number of cores sometimes.

Reply:
done